common/components/BalanceSidebar/TokenBalances/AddCustomTokenForm/AddressField.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { Input } from 'components/ui';
+import { translateRaw } from 'translations';
+import { IGenerateAddressLookup } from './AddCustomTokenForm';
+import { isValidETHAddress } from 'libs/validators';
+import { Result } from 'nano-result';
+
+interface OwnProps {
+  addressLookup: IGenerateAddressLookup;
+  onChange(address: Result<string>): void;
+}
+
+enum ErrType {
+  INVALIDADDR = 'Not a valid address',
+  ADDRTAKEN = 'A token with this address already exists'
+}
+
+interface State {
+  address: Result<string>;
+  userInput: string;
+}
+
+export class AddressField extends React.Component<OwnProps, State> {
+  public state: State = {
+    address: Result.from({ res: '' }),
+    userInput: ''
+  };
+
+  public render() {
+    const { userInput, address } = this.state;
+
+    return (
+      <label className="AddCustom-field form-group">
+        <div className="input-group-header">{translateRaw('TOKEN_ADDR')}</div>
+        <Input
+          isValid={address.ok()}
+          className="input-group-input-small"
+          type="text"
+          name="Address"
+          value={address.ok() ? address.unwrap() : userInput}
+          onChange={this.handleFieldChange}
+        />
+        {address.err() && <div className="AddCustom-field-error">{address.err()}</div>}
+      </label>
+    );
+  }
+
+  private handleFieldChange = (args: React.FormEvent<HTMLInputElement>) => {
```

Comment:
I'm not a fan of using `args` vs. the standard `e` for HTMLInput events.

Reply:
Agreed.